docusaurus/docs/migration-guides/angular-react/angular-react-custom-configuration-components.md

```diff
@@ -0,0 +1,62 @@
+---
+id: custom-config-components
+title: Custom configuration components
+sidebar_position: 4
+description: How to migrate a Grafana plugin from AngularJS with custom fields to React with custom components.
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - React
+  - ReactJS
+  - Angular
+  - migration
+---
+
+# AngularJS to React: Custom configuration components
+
+## Background
+
+The Grafana SDK provides many useful configuration components for developers like entering number ranges, thresholds, unit selections, and color selections. You can see descriptions of these elements in the [Grafana design system](https://grafana.com/developers/saga/about/overview).
+
+Some panels have more complex configuration options than others. These panels require you to build a custom component to support porting from AngularJS to React.
+
+## An Angular custom configuration example
+
+This document is focused around the `grafana-polystat-panel` as an example of a large complex plugin that was converted from AngularJS to React.
+
+The AngularJS version of `polystat` has a custom editor for composites. A composite is a synthetic metric made by grouping related metrics source from multiple queries. This allows viewing a service (or any device) as a unit for overall health.
```

Comment:
IMO we can leave out the definition of a composite here, adds too much complexity to the topic and it is not really required to understand this doc.l

Reply:
Terms that may be unfamiliar to users should be defined on first use. However, if we don't want to clutter this topic, can we link to a definition elsewhere? This one may be similar enough: https://en.wikipedia.org/wiki/Composite_measure

Reply:
I thought it would be easier to explain "why you need to create a new custom component" by using the composite editor that was in angular as a complex example.  It's really a worst case scenario - it has nothing like it and we don't provide anything in the UI components that can mimic it.